src/mongocxx/private/client_session.hh

```diff
@@ -77,7 +77,17 @@ class client_session::impl {
         std::unique_ptr<mongoc_session_opt_t, decltype(libmongoc::session_opts_destroy)> opt_t{
             libmongoc::session_opts_new(), libmongoc::session_opts_destroy};
 
-        libmongoc::session_opts_set_causal_consistency(opt_t.get(), _options.causal_consistency());
+        // If there is an actual representation for our consistency options, submit them; otherwise
+        // ignore and let the server apply its defaults (note that some of our compatibility targets
+        // lack has_value()):
```

Comment:
```suggestion
        // If there is an actual representation for our consistency options, submit them; otherwise
        // ignore and let the C driver apply its defaults (note that some of our compatibility
        // targets lack has_value()):
```
Comment is misleading: the defaults are determined by the _client_ for the given session options being set and requested, not the server. In this case, the default values are being deferred to the C driver implementation rather than explicitly forwarded by the C++ driver.